### PR TITLE
REF: share more EA methods

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/arrays/categorical.py [hunk 1]
```diff
@@ -1436,62 +1426,6 @@ def check_for_ordered(self, op):
                 "Categorical to an ordered one\n"
             )
 
-    def _values_for_argsort(self):
-        return self._codes
-
-    def argsort(self, ascending=True, kind="quicksort", **kwargs):
```
Comment (jreback) [inline pandas/core/arrays/categorical.py:1442]: where did this doc-string go?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/arrays/categorical.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/arrays/categorical.py:1442]: got rid of it, can just use parent class docstring

Comment (jreback) [reply pandas/core/arrays/categorical.py:1442]: I think for this one we actually want to keep it

Comment (jbrockmendel) [reply pandas/core/arrays/categorical.py:1434]: just pushed restoring it

Comment (jbrockmendel) [reply pandas/core/arrays/categorical.py:1434]: restored+green

**Review by jreback — APPROVED** [review]

**Pull request merged**